gateway-server/src/test/java/org/apache/knox/gateway/services/security/impl/JettySSLServiceTest.java

```diff
@@ -0,0 +1,475 @@
+
+    AliasService aliasService = createMock(AliasService.class);
+    expect(aliasService.getGatewayIdentityKeystorePassword()).andThrow(new AliasServiceException(null)).atLeastOnce();
+    expect(aliasService.getGatewayIdentityPassphrase()).andReturn(identityKeyPassphrase).atLeastOnce();
+
+    KeystoreService keystoreService = createMock(KeystoreService.class);
+
+    replayAll();
+
+    JettySSLService sslService = new JettySSLService();
+    sslService.setAliasService(aliasService);
+    sslService.setKeystoreService(keystoreService);
+
+    sslService.buildSslContextFactory(config);
+
+    fail("AliasServiceException should have been thrown");
+  }
+
+  @Test
+  public void TestBuildSslContextFactoryOnlyIdentityKeystoreNullKeystorePassword() throws Exception {
+    String basedir = System.getProperty("basedir");
+    if (basedir == null) {
+      basedir = new File(".").getCanonicalPath();
+    }
+
+    Path identityKeystorePath = Paths.get(basedir, "target", "test-classes", "keystores", "server-keystore.jks");
+    String identityKeystoreType = "jks";
+    char[] identityKeyPassphrase = "horton".toCharArray();
+    String identityKeyAlias = "server";
+    Path truststorePath = Paths.get(basedir, "target", "test-classes", "keystores", "server-truststore.jks");
+    String truststoreType = "jks";
+
+    GatewayConfig config = createGatewayConfig(false, false, identityKeystorePath, identityKeystoreType, identityKeyAlias, truststorePath, truststoreType);
+
+    AliasService aliasService = createMock(AliasService.class);
+    expect(aliasService.getGatewayIdentityKeystorePassword()).andReturn(null).atLeastOnce();
+    expect(aliasService.getGatewayIdentityPassphrase()).andReturn(identityKeyPassphrase).atLeastOnce();
+
+    KeystoreService keystoreService = createMock(KeystoreService.class);
+
+    replayAll();
+
+    JettySSLService sslService = new JettySSLService();
+    sslService.setAliasService(aliasService);
+    sslService.setKeystoreService(keystoreService);
+
+    Object result = sslService.buildSslContextFactory(config);
+    assertNotNull(result);
+    assertTrue(result instanceof SslContextFactory);
+
+    SslContextFactory sslContextFactory = (SslContextFactory) result;
+    sslContextFactory.start();
+
+    assertEquals(identityKeystorePath.toUri().toString(), sslContextFactory.getKeyStorePath());
+    assertEquals(identityKeystoreType, sslContextFactory.getKeyStoreType());
+    assertNotNull(sslContextFactory.getKeyStore());
+
+    assertNull(sslContextFactory.getTrustStorePath());
+    assertNull(sslContextFactory.getTrustStoreType());
+
+    // If the truststore is not set, by default the identity keystore is used by Jetty.
+    assertEquals(sslContextFactory.getKeyStore().size(), sslContextFactory.getTrustStore().size());
+    assertTrue(sslContextFactory.getTrustStore().containsAlias(identityKeyAlias));
+
+    verifyAll();
+
+    // Note: The key password is used if the keystore password is not set; and vice versa
```

Comment:
Seems like this comment out of place?

Reply:
Nope... it is where I wanted it since it is a side note for anyone reading the code and asking why is passed to this point.

Reply:
ok 👍